### PR TITLE
Refactor the Hero component

### PR DESCRIPTION
**Pull request by daileytj**
Fixes PXBLUE-2460.

#### Changes proposed in this Pull Request:
- Refactor the `Hero` component to accept `channelValueProps` instead of individual props to pass through to the child `ChannelValue`
- Update storybook stories to use refactored `Hero` component
- Update storybook to RN 64 (probably should have done this separately, but at least its all in one commit ¯\_(ツ)_/¯ )

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 -   `IconClass` prop for most components has been renamed to `icon` for clarification and has bee extended to support a wider variety of icon formats (see [Icons](https://github.com/pxblue/react-native-component-library/blob/master/docs/Icons.md)]).
 -   In `<Hero>` component, `ValueIconClass` prop has been renamed to `valueIcon`.
--   In `<ChannelValue>` component, `IconProps` has been replaced with separate props for `iconSize` and `iconColor`.
+-   In `<ChannelValue>` component, `IconProps` has been replaced with separate props for `iconSize` and `iconColor`, and `fontSize`, `value`, `valueIcon`, `valueColor` and `units` props have been replaced with a single `channelValueProps` prop.
```
Comment (joebochill) [inline CHANGELOG.md:19]:
```suggestion
-   In `<ChannelValue>` component, `IconProps` has been replaced with separate props for `iconSize` and `iconColor`, 
-   In `<Hero>` component, fontSize`, `value`, `valueIcon`, `valueColor` and `units` props have been replaced by `ChannelValueProps` prop, which will allow you to specify any props on the underlying `<ChannelValue>` component.
```

File: components/src/core/hero/hero.tsx [hunk 2]
```diff
@@ -83,23 +83,9 @@ export type HeroProps = ViewProps & {
...
-
-    /** A component to render for the ChannelValue child */
-    valueIcon?: IconSource;
-
-    /** Color to use for the ChannelValue text */
-    valueColor?: string;
-
-    /** Units for the ChannelValue child */
-    units?: string;
+    channelValueProps?: ChannelValueProps;
```
Comment (joebochill) [inline components/src/core/hero/hero.tsx:88]: Our convention for Pass-Through props is to TitleCase them
```suggestion
    ChannelValueProps?: ChannelValueProps;
```

File: components/src/core/hero/hero.tsx [hunk 3]
```diff
@@ -123,16 +109,12 @@ export type HeroProps = ViewProps & {
  *
  * The Hero is used to call attention to particular values of importance to the user. It includes a
  * large icon with a label and (typically) a [ChannelValue](https://pxblue-components.github.io/react-native/?path=/info/components-documentation--channel-value) item. The channel value can be configured
- * through the `value` and `units` props, or passed declaratively as a child.
+ * through the `ChannelValue` props, or passed declaratively as a child.
```
Comment (joebochill) [inline components/src/core/hero/hero.tsx:112]:
```suggestion
 * through the `ChannelValueProps` or passed declaratively as a child.
```

File: docs/Hero.md [hunk 4]
```diff
@@ -8,34 +8,30 @@ Hero items are used to call attention to particular values.
...
-| valueColor          | Color to use for the ChannelValue text          | `string`                               | no       | `text`                 |
-| units               | The units for the ChannelValue                  | `string`                               | no       |                        |
-| onPress             | A function to execute when pressed              | `function`                             | no       |                        |
-| theme               | Theme value overrides                           | `$DeepPartial<ReactNativePaper.Theme>` | no       |                        |
+| Prop Name           | Description                                | Type                                   | Required | Default                |
+| ------------------- | ------------------------------------------ | -------------------------------------- | -------- | ---------------------- |
+| label               | The text shown below the `ChannelValue`    | `string`                               | yes      |                        |
```
Comment (emclaug2) [inline docs/Hero.md:27]: Changes look good, since you're in here can you alphabetize these props?

**Review by emclaug2 — APPROVED** [review]

File: docs/Hero.md (hunk 4 above)
Comment (daileytj) [reply docs/Hero.md:27]: Yep yep, good call!

**Pull request merged**